ArduCopter/Copter.cpp

```diff
@@ -392,6 +392,18 @@ bool Copter::set_circle_rate(float rate_dps)
     return true;
 }
 
+// set desired speed (m/s). Used for scripting
+bool Copter::set_desired_speed(float speed)
+{
+    // exit if vehicle is not in auto mode
+    if (flightmode != &mode_auto) {
```

Comment:
I think we could use the is_autopilot() method here..

Reply:
I looked at doing so and generally agree, but I was not confident that `wp_nav->set_speed_xy(speed * 100.0f)` would function predictably if the mode was not AUTO.  If I was simply being overly conservative, then I'm happy to make the change!

Reply:
Change made per request.  Rebase/squashed commit containing Copter.cpp